packages/next/src/template-renderer/index.tsx

```diff
@@ -9,7 +9,10 @@ import type { ReactNode } from 'react';
 
 export type TemplateRendererProps = {
 	getTemplateData?: ( typeof QueryEngine )[ 'getTemplateData' ];
-	children: ( editorBlocks: BlockData[] ) => ReactNode;
+	children: ( args: {
+		editorBlocks: BlockData[];
+		is404: boolean;
```

Comment:
If we can get this out of `getTemplateData()` we do we also need to expose it directly?

Reply:
I am not entirely sure what you mean here. I assume you mean one of these:

Why is 404 status added to the response when we can get is404 out of `getTemplateData()`?
- Acceptance criteria. We can get rid of this flag in template renderer if @ayushnirwal says so.

Why do we need a separate function to get 404 status?
- `getTemplateData` processes a big `GetCurrentTemplate` query, which has other fields. All we need for now is the `connectedNode`, so it makes sense to have it separate.

Reply:
Sorry, it was your first guess: I was asking about the line I commented on: the `is404` param that we're passing to the children.

Can you point me to where that acceptance criteria is listed? Afaik I'm the one who specced both #150 and https://gihtub.com/rtCamp/headless/issues/451 not @ayushnirwal (though obviously if Ayush - or you - has a compelling case for why we need it, I'm happy to defer).

Reply:
From https://github.com/rtCamp/snapwp/issues/145#issuecomment-2788390101:
> This is meant to be a quick fix to add a status code to the response. The changes to the parser may be moved to Template rendering. After https://github.com/rtCamp/snapwp/pull/123

From https://github.com/rtCamp/snapwp/pull/151#issuecomment-2791810084:
> If next provides ways to attach status code we can do that.

Reply:
This comment is still valid after our re-scope. Instead of passing `is404` as a prop to the Template Renderer (which it has 0 to do with editor blocks), they should just call `const { is404 } = await getTemplateData()` inside `page.tsx` (and that's what `status-codes.md` should show )